File: FirebaseAIBridge/FirebaseAIBridge/Sources/ContentModality.swift
//
//  ContentModality.swift
//  FirebaseAIBridge
//
//  Created by Sean Chin on 30/5/25.
//
import Foundation
import FirebaseAI

@objc public enum ContentModalityObjc: Int {
    case unspecified = 0
    case text = 1
    case image = 2
    case video = 3
    case audio = 4
    case document = 5
    
    public static func from(_ modality: ContentModality) -> ContentModalityObjc {
        switch modality.rawValue {
        case ContentModality.text.rawValue:
            return .text
        case ContentModality.image.rawValue:
            return .image
        case ContentModality.video.rawValue:
            return .video
        case ContentModality.audio.rawValue:
            return .audio
        case ContentModality.document.rawValue:
            return .document
        default:
            return .unspecified
        }
    }
}

File: firebase-ai-kmp/src/commonMain/kotlin/io/github/seanchinjunkai/firebase/ai/type/ContentModality.kt
package io.github.seanchinjunkai.firebase.ai.type

public class ContentModality private constructor(public val ordinal: Int) {
    public companion object {
        public val UNSPECIFIED: ContentModality = ContentModality(0)

        public val TEXT: ContentModality = ContentModality(1)

        public val IMAGE: ContentModality = ContentModality(2)

        public val VIDEO: ContentModality = ContentModality(3)

        public val AUDIO: ContentModality = ContentModality(4)

        public val DOCUMENT: ContentModality = ContentModality(5)
    }
}
